HoloLensCommander/HoloLensCommander/Dialogs/SetAPItokenDialog.xaml.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Windows.UI.Xaml.Controls;
+
+// The Content Dialog item template is documented at https://go.microsoft.com/fwlink/?LinkId=234238
+
+namespace HoloLensCommander
+{
+    public sealed partial class SetAPItokenDialog : ContentDialog
+    {
+        private UserToken apiToken;
+
+        public SetAPItokenDialog(UserToken APItoken)
+        {
+            this.apiToken = APItoken;
+            this.DataContext = new SetAPItokenDialogViewModel(APItoken);
+            this.InitializeComponent();
+        }
+
+        private void ContentDialog_PrimaryButtonClick(ContentDialog sender, ContentDialogButtonClickEventArgs args)
+        {
+            ((SetAPItokenDialogViewModel)this.DataContext).UpdateAPItoken(this.apiToken);
+        }
+
+        private void ContentDialog_SecondaryButtonClick(ContentDialog sender, ContentDialogButtonClickEventArgs args)
```

Comment:
You should be able to get rid of this method by not specifying a click event for the secondary button. This is what the Connect to Device dialog does.